### PR TITLE
Adds automatic chat highlighting + a toggle for a sound playing when someone mentions the AI on the radio.

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes any variation of "AI" plus the AI's name and name with dots removed, be automatically highlighted in the chat for AI players.
Adds a toggle for a sound playing whenever "AI" or the AI's name is mentioned for the AI player.

## Why It's Good For The Game
Allows AI players to more easily notice people calling for their attention.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/c3fb21c3-8846-432d-a931-dea99cb7e691

</details>

## Changelog
:cl:
add: Added automatic chat highlighting for AI players that includes "AI" and the AI player's name
add: Added a toggle for a sound playing whenever the AI player hears "AI" or their name
/:cl:

### CODE REVIEW

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -1113,3 +1133,24 @@
...
+	temp_holder = replacetext(temp_holder, ")", "")
+	var/list/words = splittext(temp_holder, "|")
+	var/found_mention = FALSE
+	for(var/found_word in words)
+		if(findtext(message, found_word))
+			found_mention = TRUE
+			break
+
+	if(ai_notice_jingle && found_mention)
+		playsound_local(get_turf(src), 'sound/machines/twobeep.ogg', 50)
```
Comment (CydiaLamiales) [inline code/modules/mob/living/silicon/ai/ai.dm]:
```suggestion
		playsound_local(get_turf(src), 'sound/machines/twobeep.ogg', 25)
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -210,6 +218,7 @@
 	QDEL_NULL(doomsday_device)
 	QDEL_NULL(aiMulti)
 	QDEL_NULL(alert_control)
+	UnregisterSignal(src, COMSIG_MOVABLE_HEAR, PROC_REF(handle_hearing))
```
Comment (PowerfulBacon) [inline code/modules/mob/living/silicon/ai/ai.dm:221]: Signals are automatically unregistered on Destroy() but I guess it doesn't hurt to explicitly unregister it.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 3]
```diff
@@ -1113,3 +1133,24 @@
...
 /mob/living/silicon/ai/zMove(dir, feedback = FALSE, feedback_to = src)
 	. = eyeobj.zMove(dir, feedback, feedback_to)
+
+/mob/living/silicon/ai/proc/handle_hearing(datum/source, list/hearing_args)
+	SIGNAL_HANDLER
+	var/atom/user = hearing_args[HEARING_SPEAKER]
+	if(!ai_chat_highlights || user?.name == name)
+		return
+	var/message = hearing_args[HEARING_RAW_MESSAGE]
+	message = ai_chat_highlights.Replace(message, "<span style='color:pink'>$1</span>")
```
Comment (PowerfulBacon) [inline code/modules/mob/living/silicon/ai/ai.dm:1143]: This should use one of the new colour spans like cfc_pink

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 4]
```diff
@@ -1113,3 +1133,24 @@
...
+	hearing_args[HEARING_RAW_MESSAGE] = message
+	var/temp_holder = ai_chat_highlights.name
+	temp_holder = replacetext(temp_holder, "(", "")
+	temp_holder = replacetext(temp_holder, ")", "")
+	var/list/words = splittext(temp_holder, "|")
+	var/found_mention = FALSE
+	for(var/found_word in words)
+		if(findtext(message, found_word))
+			found_mention = TRUE
+			break
```
Comment (PowerfulBacon) [inline code/modules/mob/living/silicon/ai/ai.dm:1153]: This is a very strange way to run a regex, simply early return if regex.Find(message) returns false.

Comment (PowerfulBacon) [reply code/modules/mob/living/silicon/ai/ai.dm:1153]: https://www.byond.com/docs/ref/#/regex/proc/Find

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**